src/bio/bio_xstream.c

```diff
@@ -1529,18 +1531,21 @@ bio_xsctxt_free(struct bio_xs_context *ctxt)
 	ABT_mutex_unlock(nvme_glb.bd_mutex);
 
 	if (ctxt->bxc_thread != NULL) {
-		D_DEBUG(DB_MGMT, "Finalizing SPDK thread, tgt_id:%d",
-			ctxt->bxc_tgt_id);
+		D_DEBUG(DB_MGMT, "Finalizing SPDK thread, tgt_id:%d, skip_draining:%u",
+			ctxt->bxc_tgt_id, ctxt->bxc_skip_draining);
 
-		/* Don't drain events if spdk_subsystem_fini() timeout */
-		while (rc == 0 && !spdk_thread_is_idle(ctxt->bxc_thread))
+		/*
+		 * Don't drain events if we are asked to skip this (usually
+		 * due to an earlier error).
+		 */
+		while (!ctxt->bxc_skip_draining && !spdk_thread_is_idle(ctxt->bxc_thread))
```

Comment:
@NiuYawei, please let me know if the changes in this file matches what you suggested to me a while ago. Thanks.

Reply:
Yes, it looks good to me.

Reply:
Thanks for the confirmation.